examples/profiling/primitive_instrumentation.cpp

```diff
@@ -0,0 +1,205 @@
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int hpx_main()
+{
+    // compile the given code
+    phylanx::execution_tree::compiler::function_list snippets;
+
+    auto lra = phylanx::execution_tree::compile(
+        phylanx::ast::generate_ast(lra_code), snippets);
+
+    // print instrumentation information, if enabled
+    auto entries =
+        hpx::agas::find_symbols(hpx::launch::sync, "/phylanx/*");
+
+    // LRA arguments
+
+    blaze::DynamicMatrix<double> v1{ { 15.04, 16.74 },{ 13.82, 24.49 },
+    { 12.54, 16.32 },{ 23.09, 19.83 },{ 9.268, 12.87 },{ 9.676, 13.14 },
+    { 12.22, 20.04 },{ 11.06, 17.12 },{ 16.3 , 15.7 },{ 15.46, 23.95 },
+    { 11.74, 14.69 },{ 14.81, 14.7 },{ 13.4 , 20.52 },{ 14.58, 13.66 },
+    { 15.05, 19.07 },{ 11.34, 18.61 },{ 18.31, 20.58 },{ 19.89, 20.26 },
+    { 12.88, 18.22 },{ 12.75, 16.7 },{ 9.295, 13.9 },{ 24.63, 21.6 },
+    { 11.26, 19.83 },{ 13.71, 18.68 },{ 9.847, 15.68 },{ 8.571, 13.1 },
+    { 13.46, 18.75 },{ 12.34, 12.27 },{ 13.94, 13.17 },{ 12.07, 13.44 } };
+
+    blaze::DynamicVector<double> v2{ 1, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0,
+        1, 0, 0, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1 };
+    auto x = phylanx::ir::node_data<double>{ v1 };
+    auto y = phylanx::ir::node_data<double>{ v2 };
+    auto alpha = phylanx::ir::node_data<double>{ 1e-5 };
+    bool enable_output = true;
+
+    // Time execution
+    hpx::util::high_resolution_timer t;
+
+    // Evaluate LRA using the read data
+    auto result = lra(x, y, alpha);
+
+    // CSV Header
+    std::cout << "primitive_instance" << ","
+        << "count" << ","
+        << "time" << ","
+        << "direct_count" << ","
+        << "direct_time"
+        << std::endl;
+
+    // List of existing primitive instances
+    std::vector<std::string> existing_primitive_instances;
+    // Iterate over all primitive types
+    for (auto const& pattern :
+        phylanx::execution_tree::get_all_known_patterns())
+    {
+        // Name of the primitive
+        std::string const& name = hpx::util::get<0>(pattern);
+
+        // Retrieve all instances of this primitive type
+        for (auto const& entry : hpx::agas::find_symbols(
+                 hpx::launch::sync, "/phylanx/" + name + "#*"))
+        {
+            existing_primitive_instances.push_back(entry.first);
+        }
+    }
+
+    // Print performance data
+    for (auto const& entry :
+        retrieve_counter_data(existing_primitive_instances, hpx::find_here(),
+            std::vector<std::string>{"count/eval", "time/eval",
+                "count/eval_direct", "time/eval_direct"}))
+    {
+        std::cout << "\"" << entry.first << "\"";
+        for (auto const& counter_value : entry.second)
+        {
+            std::cout << "," << counter_value;
+        }
+        std::cout << std::endl;
+    }
+
+    auto elapsed = t.elapsed();
+
+    // Make sure all counters are properly initialized, don't reset current counter values
```

Comment:
Either you move this before retrieving the counter values or this can be removed.

Reply:
I was referring to the invocation of `reinit_active_counters()` below, sorry for being insufficiently concise.

Reply:
Fixed.